### PR TITLE
34 refactor has and belongs to many

### PR DESCRIPTION
**Pull request by JPrevost**
## Status
**READY**

#### What does this PR do?

- drops advisor remnants as they are not currently needed in app (this is included as it would have needed refactoring if not dropped)
- refactors Theses/Degree relationship to use a join table matching best practices
- refactors Theses/Departments relationship to use a join table matching best practices

#### Helpful background context (if appropriate)

http://blog.flatironschool.com/why-you-dont-need-has-and-belongs-to-many/

#### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/TI-34

#### Todo:
- [x] Tests
- [x] Documentation

#### Requires Database Migrations?
YES

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: app/models/degree_thesis.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# == Schema Information
+#
+# Table name: degree_theses
```
Comment (thatandromeda) [inline app/models/degree_thesis.rb]: Why is one of these plural and one of them singular?

File: app/models/department.rb [hunk 2]
```diff
@@ -9,7 +9,8 @@
 #
 
 class Department < ApplicationRecord
-  has_and_belongs_to_many :theses
+  has_many :department_theses
```
Comment (thatandromeda) [inline app/models/department.rb]: (same singular/plural question)

**Comment by thatandromeda** [comment]
I would like to be convinced we're not confusing future-us with the singular-vs-plural name thing (and also there's a rebase). Other than that, looks 🌈 .

File: app/models/department.rb (hunk 2 above)
Comment (JPrevost) [reply app/models/department.rb]: I oddly ran into problems unless I did it this way. I'm confident it was solvable with different variants, but I landed on not really caring enough to figure out why I couldn't use all plural or all singular without jumping through hoops. Sorry I lazied it :/

Comment (thatandromeda) [reply app/models/department.rb]: Coolio. Can you leave a comment so we're not confused in future? Then go ahead with the rebase/squash/merge thing.

**Pull request merged**